modules/gis/src/lib/binary-to-geojson.js

```diff
@@ -0,0 +1,115 @@
+export function binaryToGeoJson(data, type) {
+  const isFeature = Boolean(
+    data.featureIds || data.globalFeatureIds || data.numericProps || data.properties
+  );
+
+  // If not a feature, return only the geometry
+  if (!isFeature) {
+    return parseGeometry(data, type);
+  }
+
+  // TODO parse binary features, incl properties
+  return parseFeature(data, type);
+}
+
+function parseFeature(data, type) {
+  const geometry = parseGeometry(data, type);
+  return {type: 'Feature', geometry};
+}
+
+function parseGeometry(data, type) {
+  switch (type || parseType(data)) {
+    case 'point':
+      return pointToGeoJson(data);
+    case 'lineString':
+      return lineStringToGeoJson(data);
+    case 'polygon':
+      return polygonToGeoJson(data);
+    default:
+      throw new Error('Invalid type');
```

Comment:
If we are going to have an error message let's have a slightly better error message as this will leave an end user non-plussed. At least "invalid geometry type". If not just assert with default message.